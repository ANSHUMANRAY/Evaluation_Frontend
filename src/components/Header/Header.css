.header{
    color: #fff;
    text-align: center;
    padding-left: 10%;
    display: flex;
}