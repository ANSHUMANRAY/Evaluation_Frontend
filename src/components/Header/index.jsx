import React from 'react';
import { ThemeContext } from '../../contexts/themeContext';
import { useNavigate } from 'react-router-dom';
import './Header.css';

export default function Header() {
  const navigate = useNavigate();
  const handleClick = () => {
    navigate('/');
  };
  const {theme} = React.useContext(ThemeContext);
  return(
    <div className='header' style={{backgroundColor:`${theme}`}}>
      <h1 onClick={handleClick}>EVENTIFY</h1>
    </div>
  );
}