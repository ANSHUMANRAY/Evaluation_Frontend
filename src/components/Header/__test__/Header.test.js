import React from 'react';
import { render } from '@testing-library/react';
import Header from '../index';
import { ThemeContext } from '../../../contexts/themeContext';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

describe('Header', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ThemeContext.Provider value={{ theme: 'black', setTheme: jest.fn() }}>
        <Header />
      </ThemeContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});