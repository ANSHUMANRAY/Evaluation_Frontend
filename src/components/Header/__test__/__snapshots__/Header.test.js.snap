// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <div
    class="header"
    style="background-color: black;"
  >
    <h1>
      EVENTIFY
    </h1>
  </div>
</div>
`;
